packages/amplication-server/src/services/segmentAnalytics/segmentAnalytics.service.ts

```diff
@@ -101,3 +103,14 @@ export class SegmentAnalyticsService {
     } as TrackData);
   }
 }
+
+function parseValidUnixTimestampOrUndefined(value: string): string | undefined {
+  const timestamp = parseInt(value, 10);
+
+  // Check if the value is an integer and within a valid range for Unix timestamps
+  if (!isNaN(timestamp) && Number.isInteger(timestamp) && timestamp >= 0) {
+    return value;
+  } else {
+    return undefined;
+  }
+}
```

Comment:
Let's try to be consistent with the rest of the file making this function a private method of the service class